sdcm/utils/common.py

```diff
@@ -98,6 +98,20 @@
 SCYLLA_GCE_IMAGES_PROJECT = "scylla-images"
 
 
+class KeyBasedLock():  # pylint: disable=too-few-public-methods
+    """Class designed for creating locks based on hashable keys."""
+
+    def __init__(self):
+        self.key_lock_mapping = {}
```

Comment:
No code is cleaning this mapping...

Reply:
It may be added, yes. But where and why we will delete it?
We don't delete existing locks in other places.

Reply:
in other places we have one lock, here you are gonna have one per pod.
we should clean it when destroy pod/node, shouldn't we ?

Reply:
We destroy pod but then another appears.
In worst case we will have +1 lock - the one used for the temporary pod caused by `cluster extension` operation.

Reply:
One  lock +-  per loader pod.

o.k. forgot it's just in dynamic loader, and that it might be for any node